File: ros2_ws/src/my_cpp_pkg/src/led_panel.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/led_panel_state.hpp"
#include "my_robot_interfaces/srv/set_led.hpp"

class LedPanelNode : public rclcpp::Node
{
public:
    LedPanelNode() : Node("led_panel")
    {        
        this->declare_parameter("led_states", std::vector<bool>{false, false, false});

        ledPanelState = this->get_parameter("led_states").as_bool_array();
        service_ = this->create_service<my_robot_interfaces::srv::SetLed>("set_led", 
            std::bind(&LedPanelNode::callbackService, this, std::placeholders::_1, std::placeholders::_2));
        publisher_ = this->create_publisher<my_robot_interfaces::msg::LedPanelState>("led_panel_state", 10);    
        timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&LedPanelNode::callbackTimer, this));    
        RCLCPP_INFO(this->get_logger(), "led_panel has been started");
    }
private:    
    std::vector<bool> ledPanelState;
    rclcpp::Service<my_robot_interfaces::srv::SetLed>::SharedPtr service_;
    rclcpp::Publisher<my_robot_interfaces::msg::LedPanelState>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    void callbackService(my_robot_interfaces::srv::SetLed::Request::SharedPtr request, 
        my_robot_interfaces::srv::SetLed::Response::SharedPtr response)
    {
        ledPanelState[request->led_number - 1] = request->led_status;
        response->success = true;
    }

    void callbackTimer()
    {
        auto msg = my_robot_interfaces::msg::LedPanelState();
        for(int i = 0; i < 3; i++)
        {
            msg.led_status[i] = ledPanelState[i];
        }
        publisher_->publish(msg);
    }
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<LedPanelNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/led_panel.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import LedPanelState
from my_robot_interfaces.srv import SetLed


class LedPanelStateNode(Node):

    def __init__(self):
        super().__init__("led_panel")

        self.declare_parameter("led_states", [False, False, False])

        self.led_panel_status = self.get_parameter("led_states").value         
        self.service_ = self.create_service(SetLed, "set_led", self.callback_service)
        self.publisher_ = self.create_publisher(LedPanelState, "led_panel_state", 10)
        self.timer_ = self.create_timer(1.0, self.callback_timer)
        self.get_logger().info("led_panel has been started")

    def callback_service(self, request: SetLed.Request, response: SetLed.Response):
        self.led_panel_status[request.led_number - 1] = request.led_status
        response.success = True
        return response

    def callback_timer(self):
        msg = LedPanelState()
        for i in range(len(self.led_panel_status)):
            msg.led_status[i] = self.led_panel_status[i]
        self.publisher_.publish(msg)        


def main(args=None):
    rclpy.init(args=args)
    node = LedPanelStateNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()